cardano-cli/src/Cardano/CLI/EraBased/Commands/Transaction.hs

```diff
@@ -165,6 +167,25 @@ data TransactionCalculateMinFeeCmdArgs = TransactionCalculateMinFeeCmdArgs
   , txByronWitnessCount   :: !TxByronWitnessCount
   } deriving Show
 
+data TransactionCalculateMinFee2CmdArgs era = TransactionCalculateMinFee2CmdArgs
+  { eon                     :: !(ShelleyBasedEra era)
+  , nodeSocketPath          :: !SocketPath
+  , consensusModeParams     :: !ConsensusModeParams
+  , networkId               :: !NetworkId
+  , txBodyFile              :: !(TxBodyFile In)
+  , protocolParamsFile      :: !ProtocolParamsFile
+  , txIns                   :: ![(TxIn, Maybe (ScriptWitnessFiles WitCtxTxIn))]                 -- ^ Transaction inputs with optional spending scripts
+  , readOnlyReferenceInputs :: ![TxIn]                                                          -- ^ Read only reference inputs
+  , txinsc                  :: ![TxIn]                                                          -- ^ Transaction inputs for collateral, only key witnesses, no scripts.
+  , mValue                  :: !(Maybe (Value, [ScriptWitnessFiles WitCtxMint]))                -- ^ Multi-Asset value with script witness
+  , certificates            :: ![(CertificateFile, Maybe (ScriptWitnessFiles WitCtxStake))]     -- ^ Certificates with potential script witness
+  , withdrawals             :: ![(StakeAddress, Coin, Maybe (ScriptWitnessFiles WitCtxStake))]  -- ^ Withdrawals with potential script witness
+  , voteFiles               :: ![(VoteFile In, Maybe (ScriptWitnessFiles WitCtxStake))]
+  , proposalFiles           :: ![(ProposalFile In, Maybe (ScriptWitnessFiles WitCtxStake))]
```

Comment:
In order to use `calculateMinTxFee` function from `cardano-api`, we require `utxo` which is set up using these CLI arguments: `txIns`, `readOnlyReferenceInputs`, `txinsc`, `mValue`, `certificates`, `withdrawals`, `voteFiles` and `proposalFiles`.

Furthermore we need to connect to the node to query for `utxo`.

Reply:
It's possible this amount of information we require can be minimised, but I haven't yet figured it out.